### PR TITLE
FE-1519 - [Rainbow Component]

### PR DESCRIPTION
**Pull request by dawidzarzycki**
# Description
- Update rainbow component
- Update tests
# Screenshots / Gifs
Zrzut ekranu 2019-07-5 o 09 59 23

### CODE REVIEW

File: src/components/rainbow/rainbow.style.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        text-align: center;
+    }
+`;
+
+RainbowStyle.defaultProps = {
+  theme: baseTheme
+};
+
+// eslint-disable-next-line import/prefer-default-export
+export { RainbowStyle };
```
Comment (grabkowski) [inline src/components/rainbow/rainbow.style.js]: why not a default export?

File: src/components/rainbow/rainbow.component.js [hunk 2]
```diff
@@ -60,44 +32,25 @@ class Rainbow extends React.Component {
...
-   * Renders the component.
-   *
-   * @method render
-   * @return {Object} JSX
-   */
+  /** Renders the component. */
   render() {
     return (
-      <div className={ this.mainClasses } { ...tagComponent('rainbow', this.props) }>
+      <RainbowStyle className={ this.mainClasses } { ...tagComponent('rainbow', this.props) }>
```
Comment (grabkowski) [inline src/components/rainbow/rainbow.component.js]: couldn't it be just `this.props.className`?

File: src/components/rainbow/rainbow.component.js [hunk 3]
```diff
@@ -194,4 +147,15 @@ function generateConfig(immutableData, title) {
...
 
+Rainbow.propTypes = {
+  /** Supply a custom title for the chart. */
+  title: PropTypes.string,
+  /** Supply data for the chart. */
+  data: PropTypes.object.isRequired,
+  /** Supply a custom config object to the Highcharts. */
+  config: PropTypes.object,
+  /** Classes to apply to the component. */
+  className: PropTypes.string
```
Comment (grabkowski) [inline src/components/rainbow/rainbow.component.js]: please mark that's a legacy prop

File: src/components/rainbow/rainbow.style.js (hunk 1 above)
Comment (dawidzarzycki) [reply src/components/rainbow/rainbow.style.js]: I wanted to maintain a permanent convention so when we are in the component file we can always import style with brackets `import { RainbowStyle } from /style`

File: src/components/rainbow/rainbow.component.js [hunk 4]
```diff
@@ -2,35 +2,11 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { merge } from 'lodash';
-import tagComponent from '../../utils/helpers/tags';
-import './rainbow.scss';
+import tagComponent from '../../utils/helpers/tags/tags';
```
Comment (izakrzewska) [inline src/components/rainbow/rainbow.component.js]: import tagComponent from '../../utils/helpers/tags'

File: src/components/rainbow/rainbow.spec.js [hunk 5]
```diff
@@ -1,9 +1,8 @@
 import React from 'react';
-import TestUtils from 'react-dom/test-utils';
-import Rainbow from './rainbow';
 import Immutable from 'immutable';
 import { shallow, mount } from 'enzyme';
-import { elementsTagTest, rootTagTest } from '../../utils/helpers/tags/tags-specs';
+import Rainbow from './rainbow.component';
+import { rootTagTest } from '../../utils/helpers/tags/tags-specs/tags-specs';
```
Comment (izakrzewska) [inline src/components/rainbow/rainbow.spec.js]: same as above

**Comment by harpalsingh** [comment]
A few things to look at:

When you hover over a tile, the selected color should remain full and not have an opacity applied to it.

**Currently doing this:**

Screenshot 2019-07-08 at 14 22 52

**Is the "more info" part meant to be that far away?**

Screenshot 2019-07-08 at 14 24 31

**Label is obstructed by tile**

Screenshot 2019-07-08 at 14 25 15

**Comment by harpalsingh** [comment]
For now we will leave the implementation in accordance with what was done before, but this component will need a re-review and design to consider what we would like to do with various states

**Pull request merged**